rest_framework/serializers.py

```diff
@@ -388,10 +400,7 @@ def to_internal_value(self, data):
 
         ret = OrderedDict()
         errors = OrderedDict()
-        fields = [
-            field for field in self.fields.values()
-            if (not field.read_only) or (field.default is not empty)
-        ]
+        fields = self._to_internal_value_fields
```

Comment:
If we're making this change we may as well just do `for field in self._to_internal_value_fields` and not bother with the `fields` variable at all.